lib/ansible/plugins/lookup/ini.py

```diff
@@ -31,6 +31,12 @@
         default: False
         type: boolean
         description: Flag to indicate if the key supplied is a regexp.
+      re_section:
+        default: False
+        type: boolean
+        description:
+          - Flag to indicate if the section name supplied is a regexp.
+          - Requires the C(re) flag to be set to C(True).
```

Comment:
Why are you going to require `re` to be `True`? I feel like that restricts this usefulness of this feature. For example, I could see someone wanting to do something similar to this:

```
    - debug:
        msg: "{{ item }}"
      with_ini:
        - 'user section=da* file=users.ini re_section=True'
```

I think that should be supported.